File: api/HomeBudget.API/HomeBudget.Application/DTOs/CategoriaDtos/CategoriaRequestDto.cs
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.CategoriaDtos
{
    public class CategoriaRequestDto
    {
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public TipoCategoria Finalidade { get; set; }
        public DateTime DataCriacaoModificacao { get; set; }
    }
}

File: web/src/interfaces/CategoriaDtos/CategoriaRequestDto.ts
import { TipoCategoria } from "../../enums/TipoCategoriaEnum";

export class CategoriaRequestDto {
  nome: string = "";
  descricao: string = "";
  finalidade: number = TipoCategoria.Despesa;
  dataCriacaoModificacao: string = new Date().toISOString();

  constructor(
    nome: string = "",
    descricao: string = "",
    finalidade: number = TipoCategoria.Despesa,
    dataCriacaoModificacao: string = new Date().toISOString(),
  ) {
    this.nome = nome;
    this.descricao = descricao;
    this.finalidade = finalidade;
    this.dataCriacaoModificacao = dataCriacaoModificacao;
  }
}
